src/components/views/dialogs/HostSignupDialog.tsx

```diff
@@ -0,0 +1,266 @@
+                        "Are you sure you wish to abort creation of the host? The process cannot be continued.",
+                    ),
+                    button: _t("Abort"),
+                    onFinished: result => {
+                        if (result) {
+                            return this.closeDialog();
+                        }
+                    },
+                },
+            );
+        }
+    }
+
+    private sendMessage = (message: IPostmessageResponseData) => {
+        this.iframeRef.current.contentWindow.postMessage(message, this.config.url);
+    }
+
+    private async sendAccountDetails() {
+        const openIdToken = await MatrixClientPeg.get().getOpenIdToken();
+        if (!openIdToken || !openIdToken.access_token) {
+            this.setState({
+                completed: true,
+                error: _t("Failed to connect to your homeserver. Please close this dialog and try again."),
+            });
+            return;
+        }
+        this.sendMessage({
+            action: PostmessageAction.HostSignupAccountDetails,
+            account: {
+                accessToken: await MatrixClientPeg.get().getAccessToken(),
+                name: OwnProfileStore.instance.displayName,
+                openIdToken: openIdToken.access_token,
+                serverName: await MatrixClientPeg.get().getDomain(),
+                userLocalpart: await MatrixClientPeg.get().getUserIdLocalpart(),
+                termsAccepted: true,
+            },
+        });
+    }
+
+    private onAccountDetailsDialogFinished = async (result) => {
+        if (result) {
+            return this.sendAccountDetails();
+        }
+        return this.closeDialog();
+    }
+
+    private onAccountDetailsRequest = () => {
+        const termsDialog = this.config.termsDialog;
+        const textComponent = (
+            <>
+                <p>
+                    {termsDialog.text}
+                </p>
+                <p>
+                    {_t("Learn more in our")}&nbsp;
+                    <a href={termsDialog.privacyPolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.privacyPolicy.text}
+                    </a>,&nbsp;
+                    <a href={termsDialog.termsOfService.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.termsOfService.text}
+                    </a> {_t("and")}&nbsp;
+                    <a href={termsDialog.cookiePolicy.href} target="_blank" rel="noreferrer noopener">
+                        {termsDialog.cookiePolicy.text}
+                    </a>.
+                </p>
+            </>
+        );
+        Modal.createDialog(
+            QuestionDialog,
+            {
+                title: termsDialog.title,
+                description: textComponent,
+                button: termsDialog.acceptText,
+                onFinished: this.onAccountDetailsDialogFinished,
+            },
+        );
```

Comment:
There is nothing here which definitively says that the user's Access Token will be sent to a third party granting that party near-unlimited permissions over their account indefinitely.

Reply:
This particular text is supposed to be in the configured `termsDialog.text`, with links to the various policy documents detailing how the access is used.

Do you think this part needs a non-configurable text which the organization hosting Element cannot customize (from config anyway)?

Reply:
Definitely, especially as I see no way for the configured text to be internationalised. This is potentially going to be totally inaccessible to non-English speakers.

Reply:
We need to have some configurable text here, otherwise this setup wizard will be locked to one hosting provider only, which is not the intention. I'll talk with design and respective project owners to see what we should do here, if we could provide a configurable part and an in-app string.

Just as a reminder, we don't currently (as Element) provide terms of service, cookie policy or privacy policy documents in other languages than English. I'm under the impression this would be difficult to do from a legal sense. The same might be true for any wording here when asking permission from users. What if a community translator gets it wrong for example?

Reply:
Strings are now translatable.